core/variant/variant_parser.cpp

```diff
@@ -1832,6 +1866,17 @@ Error VariantWriter::write(const Variant &p_variant, StoreStringFunc p_store_str
 
 		} break;
 
+		// Do not really store these, but ensure that assignments are not empty.
+		case Variant::RID: {
+			p_store_string_func(p_store_string_ud, "RID()");
```

Comment:
Could  change this to only store `RID()` for the case where it is `RID(0)`, but unsure, elsewhere this parsing matches the general GDScript syntax, as mentioned in the main comment this simply avoids empty entries, as no serialization was done at all before, so I suggest approaching this separately later if desired